bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -1571,6 +1627,54 @@ public AddImageProcess makeAddImageProcess(String timeZone, boolean addUnallocSp
 	}
 
 	/**
+	 * Gets the the datasource obj ids for the given device_id
+	 *
+	 * @param deviceId device_id
+	 *
+	 * @return A list of the data source object_id for the given device_id for
+	 *         the case.
+	 *
+	 * @throws TskCoreException if there is a problem getting the data source
+	 *                          obj ids.
+	 */
+	public List<Long> getDataSourceObjIds(String deviceId) throws TskCoreException {
+
+		// check cached map first
+		if (deviceIdToDatasourceObjId.containsKey(deviceId)) {
```

Comment:
This is not thread-safe because the map is not guarded by any lock, nor is it a concurrent map class. Rather than making this thread safe, let's just not do this optimization for now. There will be very few rows in the data_source_info table, so it might actually be more costly to control access to the collection than to just execute the query.